torchx/lib/torchx.ex

```diff
@@ -17,4 +19,71 @@ defmodule Torchx do
   """
   def device_count(:cuda), do: NIF.cuda_device_count()
   def device_count(_), do: raise("Only CUDA devices can be counted for now.")
+
+  # LibTorch API bindings
+
+  ## Creation
+
+  def arange(from, to, step \\ 1, opts \\ []) do
+    type = opts[:type] || Nx.Type.infer([from, to])
+    device = opts[:device] || :cpu
+
+    NIF.arange(from, to, step, torch_type(type), torch_device(device))
+    |> wrap_with_device()
+  end
+
+  ## Operations
+
+  def tensordot(left_ref, right_ref, left_axes, right_axes) do
+    NIF.tensordot(
+      left_ref,
+      right_ref,
+      left_axes,
+      right_axes
+    )
+    |> wrap_with_device()
+  end
```

Comment:
Similarly this function would look like this:

```suggestion
  def tensordot(left_ref, right_ref, left_axes, right_axes) do
    {device, refs} = to_refs([left_ref, right_ref])
    NIF.tensordot(
      left_ref,
      right_ref,
      left_axes,
      right_axes
    ) |> wrap_device(device)
  end
  
  defp to_refs([{device, ref} | t]) do
    refs =
      Enum.map(t, fn
        {^device, ref} -> ref
        {other_device, _ref} ->
          raise "cannot perform operations on across devices: #{inspect(device)} and #{inspect(other_device)}
      end)
      
    {device, [ref | refs]}
  end
```

This way we would always check the device and always know the device upfront to wrap in. WDYT?